libkineto/src/RoctracerActivityInterface.h

```diff
@@ -0,0 +1,178 @@
+#include <roctracer_ext.h>
+#include <roctracer_roctx.h>
+
+#endif
+#include <functional>
+#include <list>
+#include <memory>
+#include <set>
+#include <vector>
+#include <map>
+#include <unordered_map>
+#include <deque>
+
+
+namespace KINETO_NAMESPACE {
+
+using namespace libkineto;
+
+class ApiIdList
+{
+public:
+  ApiIdList();
+  bool invertMode() { return m_invert; }
+  void setInvertMode(bool invert) { m_invert = invert; }
+  void add(std::string apiName);
+  void remove(std::string apiName);
+  bool loadUserPrefs();
+  bool contains(uint32_t apiId);
+  const std::unordered_map<uint32_t, uint32_t> &filterList() { return m_filter; }
+
+private:
+  std::map<std::string, uint32_t> m_ids;
+  std::unordered_map<uint32_t, uint32_t> m_filter;
+  bool m_invert;
+};
+
+struct roctracerRow {
+  roctracerRow(uint64_t id, uint32_t domain, uint32_t cid, uint32_t pid
+             , uint32_t tid, uint64_t begin, uint64_t end)
+    : id(id), domain(domain), cid(cid), pid(pid), tid(tid), begin(begin), end(end) {}
+  uint64_t id;  // correlation_id
+  uint32_t domain;
+  uint32_t cid;
+  uint32_t pid;
+  uint32_t tid;
+  uint64_t begin;
+  uint64_t end;
+};
+
+struct kernelRow : public roctracerRow {
+  kernelRow(uint64_t id, uint32_t domain, uint32_t cid, uint32_t pid
+          , uint32_t tid, uint64_t begin, uint64_t end
+          , const void *faddr, hipFunction_t function
+          , unsigned int gx, unsigned int gy, unsigned int gz
+          , unsigned int wx, unsigned int wy, unsigned int wz
+          , size_t gss, hipStream_t stream)
+    : roctracerRow(id, domain, cid, pid, tid, begin, end), functionAddr(faddr)
+    , function(function), gridX(gx), gridY(gy), gridZ(gz)
+    , workgroupX(wx), workgroupY(wy), workgroupZ(wz), groupSegmentSize(gss)
+    , stream(stream) {}
+  const void* functionAddr;
+  hipFunction_t function;
+  unsigned int gridX;
+  unsigned int gridY;
+  unsigned int gridZ;
+  unsigned int workgroupX;
+  unsigned int workgroupY;
+  unsigned int workgroupZ;
+  size_t groupSegmentSize;
+  hipStream_t stream;
+};
+
+struct copyRow : public roctracerRow {
+  copyRow(uint64_t id, uint32_t domain, uint32_t cid, uint32_t pid
+             , uint32_t tid, uint64_t begin, uint64_t end
+             , const void* src, const void *dst, size_t size, hipMemcpyKind kind
+             , hipStream_t stream)
+    : roctracerRow(id, domain, cid, pid, tid, begin, end)
+    , src(src), dst(dst), size(size), kind(kind), stream(stream) {}
+  const void *src;
+  const void *dst;
+  size_t size;
+  hipMemcpyKind kind;
+  hipStream_t stream;
+};
+
+struct mallocRow : public roctracerRow {
+  mallocRow(uint64_t id, uint32_t domain, uint32_t cid, uint32_t pid
+             , uint32_t tid, uint64_t begin, uint64_t end
+             , const void* ptr, size_t size)
+    : roctracerRow(id, domain, cid, pid, tid, begin, end)
+    , ptr(ptr), size(size) {}
+  const void *ptr;
+  size_t size;
+};
+
+
+class RoctracerActivityInterface {
```

Comment:
How about RoctracerActivityApi to avoid overloading interface? (I can change the cupti one).

Reply:
I am in the middle of refactoring those classes.  Can we just defer the name change until after that?